src/exo/LoopIR_scheduling.py

```diff
@@ -636,24 +636,38 @@ def szop(op, lhs, rhs):
     ir, fwd_repl = fwd(loop_cursor)._child_node("hi")._replace(hi_o)
     fwd = _compose(fwd_repl, fwd)
 
-    # wrap body in inner loop
-    def inner_wrapper(body):
-        return LoopIR.For(
-            sym_i,
-            LoopIR.Const(0, T.index, srcinfo),
-            hi_i,
-            body,
-            LoopIR.Seq(),
-            None,
-            srcinfo,
+    wrapped_inner_loop = True
+    try:
+        Check_ExprEqvInContext(
+            proc, hi_i, [loop], LoopIR.Const(1.0, T.index, loop.srcinfo), [loop]
```

Comment:
Wait I'm a bit confused, why is this hardcoding 1.0? Is this a workaround for the specific pattern that you're looking at?

Reply:
I'm specifically trying to avoid generating loops of size 1.

Reply:
Why is that a problem?

Reply:
I think you want scheduling operations to always generate the same structure. Especially, that a lot of our implementations rely on that. If you are working with a case where you need this, you can probably call `rewrite_expr(proc, outer_loop.hi(), "1")`, then `unroll_loop(proc, outer_loop)`.

Reply:
Yeah I second that

Reply:
@SamirDroubi I don't love that approach because `unroll_loop` will break all forwarding to code within it. In particular, a lot of Halide code is loop nest around an assign statement, and I would like to maintain forwarding for the assign statement.

Reply:
Ugh you are right. Could we make `unroll_loop` forward in that edge case since it might be a desirable behavior in other cases as well? I actually have a bunch of cases where I unroll loops of size 1 as well.

Reply:
Will it be crazy to forward the loop body to the first unrolled loop body?

Reply:
Yeah, there's no perfect solution. I think we should minimize surprise in general.

Reply:
divide_with_recompute generating different loop structure for hi == 1 is very surprising. We should weigh that with how surprising the unroll forwarding behavior is.

Reply:
Oh, I guess one solution is rather than modifying the schedulign operation, I could build a lightweight wrapper around `divide_with_recompute` which will only perform the operation if it generates a non-trivial loop by introspecting the inner loop's bound after applying the operation.

Reply:
Instead of using `unroll_loop`, you could also use `remove_loop` whose forwarding behavior is what you want, but you might need to slightly change its implementation to handle this corner case (hi - lo = 1). In this case, you don't need to verify idempotency and you can immediately do the transform.

Although, now `remove_loop` would be correct for two different reasons rather just one.

Reply:
Changing `remove_loop` will be super hacky too. Generally I think we shouldn't hack in corner cases to scheduling ops implementation

Reply:
Yeah I considered modifying `remove_loop`, but I think I agree with Yuka about keeping primitives as general/simple as possible.